src/klein/_request_compat.py

```diff
@@ -86,7 +86,7 @@ def uri(self):
         if path and path[0] == u"/":
             path = path[1:]
 
-        return URL.fromText(u"{}://{}/{}".format(scheme, netloc, path))
+        return DecodedURL.fromText(u"{}://{}/{}".format(scheme, netloc, path))
```

Comment:
I think the preferred API for this might be `hyperlink.parse`, but without updated docs it's hard to say.  This is definitely not a blocker.

Reply:
Yeah, that function is not in https://hyperlink.readthedocs.io/en/latest/api.html#creation AFAICT.